code/modules/virology/virology_goals.dm

```diff
@@ -0,0 +1,171 @@
+	var/goal_symptom //Type path of the symptom
+	var/goal_symptom_name
+	var/goal_property
+	var/goal_property_text
+	var/goal_property_value
+
+/datum/virology_goal/propertysymptom/New()
+	var/type = pick(subtypesof(/datum/symptom))
+	var/datum/symptom/S = new type()
+	goal_symptom = S.type
+	goal_symptom_name = S.name
+	goal_property = pick("resistance", "stealth", "stage_rate", "transmittable")
+	if(goal_property == "stage_rate")
+		goal_property_text = "stage rate"
+	else
+		goal_property_text = goal_property
+	goal_property_value = rand(-18,11)
+	if(goal_property == "resistance")
+		goal_property_value += S.resistance
+	else if(goal_property == "stealth")
+		goal_property_value += S.stealth
+	else if(goal_property == "stage_rate")
+		goal_property_value += S.stage_speed
+	else
+		goal_property_value += S.transmittable
+	qdel(S)
+
+/datum/virology_goal/propertysymptom/get_report()
+	return {"<b>Effects of [goal_symptom_name] symptom and level [goal_property_value] [goal_property_text]</b><br>
+	Viral samples with a specific symptom and properties are required to study the effects of this symptom in various conditions. We need you to deliver [delivery_goal]u of viral samples containing the [goal_symptom_name] symptom and with the [goal_property_text] property at level [goal_property_value] along with 3 other symptoms to us through the cargo shuttle.
+	<br>
+	-Nanotrasen Virology Research"}
+
+/datum/virology_goal/propertysymptom/get_ui_report()
+	return {"Viral samples with a specific symptom and properties are required to study the effects of this symptom in various conditions. We need you to deliver [delivery_goal]u of viral samples containing the [goal_symptom_name] symptom and with the [goal_property_text] property at level [goal_property_value] along with 3 other symptoms to us through the cargo shuttle."}
+
+/datum/virology_goal/propertysymptom/check_completion(list/datum/reagent/reagent_list)
+	. = FALSE
+	var/datum/reagent/blood/BL = locate() in reagent_list
+	if(BL)
+		if(BL.data && BL.data["viruses"])
+			for(var/datum/disease/advance/D in BL.data["viruses"])
+				if(D.symptoms.len < 4) //We want 3 other symptoms alongside the requested one
+					continue
+				var/properties = D.GenerateProperties()
+				var/property = properties[goal_property]
+				if(!(property == goal_property_value))
+					continue
+				for(var/datum/symptom/S in D.symptoms)
+					if(S.type == goal_symptom)
+						delivered_amount += BL.volume
+						if(delivered_amount >= delivery_goal)
+							delivered_amount = delivery_goal
+							completed = TRUE
+							return TRUE
+					else
+						continue
+
+/datum/virology_goal/virus
+	name = "Specific Viral Sample Request (Non-Stealth)"
+	var/list/goal_symptoms = list() //List of type paths of the symptoms, we could go with a diseaseID here instead a list of symptoms but we need the list to tell the player what symptoms to include
+
+/datum/virology_goal/virus/New()
```

Comment:
Would it not work anymore if this was on `Initialize()` intead of `New()`?

Reply:
Could you please reply on this instead of resolving it?

Reply:
iam guessing i can resolve this, right?

Reply:
yes you can resolve this, `Initialize` is only available on atoms and not stuff subtypes directly off of datums